dlls/dxgi/swapchain.c

```diff
@@ -636,12 +673,47 @@ void qemu_dxgi_swapchain_ResizeBuffers(struct qemu_syscall *call)
 {
     struct qemu_dxgi_swapchain_ResizeBuffers *c = (struct qemu_dxgi_swapchain_ResizeBuffers *)call;
     struct qemu_dxgi_swapchain *swapchain;
+    uint64_t *host_buffers;
+    IDXGISurface1 *buffer;
+    UINT i;
 
-    WINE_FIXME("Unverified!\n");
     swapchain = QEMU_G2H(c->iface);
 
     c->super.iret = IDXGISwapChain1_ResizeBuffers(swapchain->host, c->buffer_count, c->width, c->height,
             c->format, c->flags);
+
+    if (FAILED(c->super.iret))
+        return;
+
+    /* if bufferCount == 0, then the existing buffer count is preserved, so we need to see what that is */
+    if (c->buffer_count == 0) {
+        DXGI_SWAP_CHAIN_DESC desc;
+
+        c->super.iret = IDXGISwapChain1_GetDesc(swapchain->host, &desc);
+        if (FAILED(c->super.iret)) {
+            WINE_ERR("failed to get swapchain desc");
+            return;
+        }
+        c->buffer_count = desc.BufferCount;
+    }
+
+    host_buffers = HeapAlloc(GetProcessHeap(), 0, c->buffer_count * sizeof(*host_buffers));
+    if (!host_buffers)
```

Comment:
This will crash later on in an out of memory exception. Since we are a library we should try to gracefully handle it and return an error code to the calling application. (Which will then presumably crash, but that's not our choice to make)

Reply:
NB: In situations where I am too lazy to handle OOM errors or handling it would be so complicated that I don't trust the OOM codepath without testing it I prefer a WINE_ERR over a WINE_WARN.

The general idea of the Wine log severities:
TRACE: Something normal going on
WARN: An error condition that we believe was correctly handled. E.g. invalid parameter passed by the app
FIXME: Unimplemented functionality / unknown condition. Probably a bug
ERR: Something that we didn't expect to happen at all. A call that was supposed to succeed failed, etc.

FIXME and ERR are printed by default, TRACE and WARN are not. But for debugging a warn+all log is usually a good start as it will point out stuff like possible application misbehaviour, file not found issues, etc. Very often a WARNed about misbehaviour is a follow-up problem where we gave the application some bad or unexpected result.

Reply:
FWIW this was copied directly from `qemu_dxgi_swapchain_create`. I'll change it to `WINE_ERR` though.

Reply:
Or maybe I should just pull this out into a separate function? Seems I was a bit lazy when I wrote this...

Reply:
I've pulled this out to a separate function, similar to the guest-side bits, and also fixed it to use `WINE_ERR` when an error happens and handle errors properly.

Reply:
Hmm I did double check if I made the mistake somewhere, so not to blame you for my own sloppiness. I guess I was double sloppy...